ehrapy/api/io/read.py

```diff
@@ -307,6 +332,46 @@ def _check_datafile_present_and_download(path: Union[str, Path], backup_url=None
 
         return True
 
+    @staticmethod
+    def _set_index(df: pd.DataFrame, index_column: Union[str, Optional[int]]) -> pd.DataFrame:
+        """Try to set the index, if any given by the index_column parameter."""
+        column_names = list(df.columns)
+        if isinstance(index_column, str):
+            df = df.set_index(index_column)
+        elif isinstance(index_column, int):
+            df = df.set_index(column_names[index_column])
+        else:
+            if "patient_id" == column_names[0]:
+                df = df.set_index("patient_id")
+            else:
+                warnings.warn(
+                    "Did not found patient_id column at column 0 and no index column was passed. Using default, numerical indices instead!",
+                    IndexColumnWarning,
+                )
+        return df
+
+    @staticmethod
+    def move_columns_to_obs(df: pd.DataFrame, columns_obs_only: Optional[List[Union[str]]]) -> BaseDataframes:
+        """
+        Move the given columns from the original dataframe (and therefore X) to obs. By doing so, those values will not get lost
+        and will be stored in obs, but will not appear in X. This may be useful for textual values like free text.
+        """
+        if columns_obs_only:
+            try:
+                obs = df[columns_obs_only].copy()
+                obs = obs.set_index(df.index.map(str))
+                df = df.drop(columns_obs_only, axis=1)
+            # TODO Key error traceback still prints, have no idea why
```

Comment:
@Zethson could need your help/input here: Despite "catching" the `KeyError` if a misspelled column is passed and raising the custom `ColumnNotFoundError`, the traceback of the `KeyError` is still printed and therefore still raised somehow. I can't see why this is the case. Maybe I'm missing something obvious here.

Reply:
https://paste.ofcode.org/P28YFe8L75vxVMryXwLanR